libs/utils/energy_model.py

```diff
@@ -1152,3 +1154,404 @@ def f(input_row):
             return ret
 
         return inputs.apply(f, axis=1)
+
+###########################################
+# Sanity Checks helpers for Energy Models #
+###########################################
+
+# The utilization rate limit to switch to the overutilization mode
+OVERUTILIZED_RATE = 0.8
+
+def _get_powers_list(node):
+    """
+    Get the list of powers for a given node.
+    :param node: A node of the energy model. It can be a cluster or cpu.
+    :type node: EnergyModelNode
+
+    :returns: The list of powers associates with the given node.
+    """
+    powers = []
+    for freq in node.active_states.keys():
+        powers.append(node.active_states[freq].power)
+    return powers
+
+def _get_capacities_list(node):
+    """
+    Get the list of capacities for a given node.
+    :param node: A node of the energy model.
+    :type node: EnergyModelNode
+
+    :returns: The list of capacities associates with the given node.
+    """
+    cap = []
+    for freq in node.active_states.keys():
+        cap.append(node.active_states[freq].capacity)
+    return cap
+
+def _is_list_strickly_increasing(l):
```

Comment:
`pandas.Series` has an `is_monotonic_increasing` property, so that could be turned into:
```python
    def _is_list_monotonic_increasing(l):
        return pd.Series(l).is_monotonic_increasing
```
That brings in a conversion to a Series though (dunno how smart the converstion is). The Numpy way might be a bit better:
```python
    def _is_list_monotonic_increasing(l):
        return all(diff > 0 for diff in np.diff(l))
```

Reply:
I would argue that a good old loop if more efficient that those conversions that will require memory re-allocation and all ...

Reply:
Up to you, the more I use Python the more I get the feeling that loops are so Y2K :D